coordinator/src/main/scala/filodb.coordinator/ShardAssignmentStrategy.scala

```diff
@@ -53,16 +57,26 @@ class K8sStatefulSetShardAssignmentStrategy extends ShardAssignmentStrategy with
       .map(host => InetAddress.getByName(host).getHostName)
       .orElse(Some(InetAddress.getLocalHost.getHostName))
       .map(name => if (name.contains(".")) name.substring(0, name.indexOf('.')) else name)
-      .flatMap(hostName => pat.findFirstIn(hostName).map(ordinal => (hostName, -Integer.parseInt(ordinal))))
-
+      .flatMap(hostName =>
+        Some((hostName, pat.findFirstIn(hostName).map(ordinal => -Integer.parseInt(ordinal)).getOrElse(-1))))
 
 
   override def shardAssignments(coord: ActorRef,
-                                dataset: DatasetRef,
-                                resources: DatasetResourceSpec,
-                                mapper: ShardMapper): Seq[Int] =
-     getOrdinalFromActorRef(coord) match {
-        case Some((hostName, ordinal)) =>
+                       dataset: DatasetRef,
+                       resources: DatasetResourceSpec,
+                       mapper: ShardMapper): Seq[Int] =
+    shardAssignmentsWithRetry(coord, dataset, resources, mapper, currentAttempt = 1, maxAssignmentAttempts)
+
+
+  // scalastyle:off method.length
+  private def shardAssignmentsWithRetry(coord: ActorRef,
```

Comment:
Can you refactor the ordinal discovery with retry into a separate method and reuse from `FiloDbClusterDiscovery` class as well ? This exact capability will be needed there as well.

Reply:
I don't see a clear way to reuse for following reasons. 

1. ``FiloDbClusterDiscovery`` doesn't perform host lookup by actor name
2. Only place we of a host lookup in  ``FiloDbClusterDiscovery`` is here, which looks up local host name and not a remote actor hostname. 

```  lazy val ordinalOfLocalhost: Int = {
    if (settings.localhostOrdinal.isDefined) settings.localhostOrdinal.get
    else {
      val fullHostname = InetAddress.getLocalHost.getHostName
      val hostName = if (fullHostname.contains(".")) fullHostname.substring(0, fullHostname.indexOf('.'))
      else fullHostname

      val pat = "-\\d+$".r
      // extract ordinal at the end from hostname such as filodb-host-0, filodb-host-10
      val ordinal = pat.findFirstIn(hostName).map(ordinal => -Integer.parseInt(ordinal))
        .getOrElse(throw new IllegalArgumentException(s"hostname $hostName does not appear like a stateful sets host"))
      ordinal
    }
  }```

``ShardAssignmentStrategy`` has predefined methods used by our existing Clustering based up Akka and the implementation is around these methods.